openverse_catalog/dags/providers/provider_csv_load_scripts/inaturalist/taxa.sql

```diff
@@ -3,51 +3,131 @@
+
+Integrating data from iNaturalist taxonomy table with Catalog of Life name data.
+
+Title:
+    - If the scientific name has one or more English vernacular names, collect them into
+      the title, separated by commas.
+    - If not, there are a few taxa that are very common where I googled an English name
+      and added it manually, so use that.
+    - Otherwise, use the iNaturalist name. There are a few where the name is "Not
+      assigned" but we're going to filter those records out & drop associated photos.
+
+Tags:
+    - If there are non-English vernacular names, put them in the tags.
+    - Put the titles of ancestors in the tags.
+    - If the title of a specific taxa is a vernacular name, put the iNaturalist name in
+      the tags.
+
+Representing tags in this way to be consistent with python processing
+"_enrich_tags":
+https://github.com/WordPress/openverse-catalog/blob/337ea7aede228609cbd5031e3a501f22b6ccc482/openverse_catalog/dags/common/storage/media.py#L265
+TO DO: consider normalizing image tags to a separate table instead.
+
+Could also consider declaring types and using row to json to replace
+repeated json build statements, for example
+create type openverse_tag as (name varchar(255), provider varchar(255));
 */
 
+/* ********** Load raw iNaturalist Data ********* */
 DROP TABLE IF EXISTS inaturalist.taxa;
-COMMIT;
-
 CREATE TABLE inaturalist.taxa (
     taxon_id integer,
     ancestry character varying(255),
     rank_level double precision,
     rank character varying(255),
     name character varying(255),
-    active boolean,
-    ancestor_names text
+    active boolean /*,
+    ancestor_names text */
 );
-COMMIT;
-
 SELECT aws_s3.table_import_from_s3('inaturalist.taxa',
     'taxon_id, ancestry, rank_level, rank, name, active',
     '(FORMAT ''csv'', DELIMITER E''\t'', HEADER, QUOTE E''\b'')',
     'inaturalist-open-data',
     'taxa.csv.gz',
     'us-east-1');
-
 ALTER TABLE inaturalist.taxa ADD PRIMARY KEY (taxon_id);
 COMMIT;
 
-WITH aggregated AS
+/* ********** Integrate Catalog of Life Data ********* */
+drop table if exists inaturalist.taxa_with_vernacular;
+create table inaturalist.taxa_with_vernacular as
+(
+    with
+    catalog_of_life as
+    (
+        SELECT
+            cast(md5(n.scientificname) as uuid) as md5_scientificname,
+            string_agg(DISTINCT
+                case
+                when v.name_language = 'eng' then v.taxon_name
+                end,
+                ', ') name_english,
+            '{"name": "'||string_agg(DISTINCT
+                case
+                when v.name_language <> 'eng' then v.taxon_name
+                end,
+                '", "provider": "inaturalist"}, {"name": "')
+            ||', "provider": "inaturalist"}' tags_nonenglish_vernacular
+        FROM inaturalist.col_name_usage n
+            INNER JOIN inaturalist.col_vernacular v on v.taxonid = n.id
+        where length(n.id) <= 10
+        group by cast(md5(n.scientificname) as uuid)
```

Comment:
Simple SQL literacy question, could this just be `group by 1`?

Reply:
This is such a great question! Yeah, I feel like there are some dialects or configurations where that doesn't work and others where it's more performant, but I think our postgres is one of the latter, so yeah, I'll totally change it.

Reply:
Nice! I was mostly thinking about not having to declare the same functions on the column, but it'd be even better if that is more performant 🏎️